extensions/generators/npm.py

```diff
@@ -0,0 +1,29 @@
+import json
+
+from conan import ConanFile
+from conan.tools.files import copy, mkdir, save
+from pathlib import Path
+
+
+class npm:
+    def __init__(self, conanfile: ConanFile):
+        self._conanfile = conanfile
+
+    def generate(self):
+        if self._conanfile.settings.os != "Emscripten":
+            self._conanfile.output.error("Can only deploy to NPM when build for Emscripten")
```

Comment:
I would even throw an exception here. If you are executing this code, it means it was explicitely requested to generated the npm description, but right now this cannot happen, so for me this is a critical failure.

Reply:
Yes, agreed, will change it